dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryImpl.java

```diff
@@ -226,15 +257,27 @@ protected String getLanguageCodeAndCountry(long id, String langId) {
 		return language.getLanguageCode() + "_" + language.getCountryCode();
 	}
 
-    private static Language defaultLanguage;
-
 	@Override
     protected Language getDefaultLanguage () {
+
         if (defaultLanguage == null) {
-            defaultLanguage = getLanguage (Config.getStringProperty("DEFAULT_LANGUAGE_CODE"), Config.getStringProperty("DEFAULT_LANGUAGE_COUNTRY_CODE"));
-            if (defaultLanguage.getId() == 0)
-                defaultLanguage = createDefaultLanguage();
+
+        	synchronized (this) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this dangerous instance of double-checked locking. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2168)